### PR TITLE
Add tests for profile_controller.rb

### PR DESCRIPTION
**Pull request by gr1d99**
# Description 
This PR adds test for `profile_controller.rb`

## Corresponding Issue
[#1174](https://github.com/ifmeorg/ifme/issues/1174)

# Screenshots

# Test Coverage

✅

### CODE REVIEW

File: spec/controllers/profile_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+describe ProfileController do
+  describe 'GET #index' do
+    context 'when user is not signed in' do
+      before { get :index, params: { 'uid': 'user-id' } }
+      it { expect(response).to redirect_to(new_user_session_path) }
+    end
+
+    context 'when user is logged in' do
```
Comment (julianguyen) [inline spec/controllers/profile_controller_spec.rb]: We should combine the two duplicate `when user is logged in` contexts

Comment (julianguyen) [inline spec/controllers/profile_controller_spec.rb]: We want to test when a user is logged in and looking at their own profile and when a user is logged in a looking at someone else's profile

File: app/controllers/profile_controller.rb [hunk 2]
```diff
@@ -9,13 +9,13 @@ def index
     user = current_user if user.nil?
 
     # Determine how the profile should be displayed based on the user_id
-    if user == current_user
-      @stories = Kaminari.paginate_array(get_stories(current_user, false))
+    @stories = if user == current_user
+                 Kaminari.paginate_array(get_stories(current_user, false))
```
Comment (hmasila) [inline app/controllers/profile_controller.rb]: The two statements inside the conditional are similar. 
Maybe we could have

```ruby
if user == current_user || current_user.allies_by_status(:accepted).include?(user)
  @stories = Kaminari.paginate_array(get_stories(user, false)).page(params[:page])
end
```
better enough, we could put `user == current_user || current_user.allies_by_status(:accepted).include?(user)` in a variable

Comment (hmasila) [reply app/controllers/profile_controller.rb]: From my suggestion my aim was to do away with the elseif block. That is why I did
```ruby
if user == current_user || current_user.allies_by_status(:accepted).include?(user)
  @stories = Kaminari.paginate_array(get_stories(user, false)).page(params[:page])
end
```

We also don't need a separate method for the pagination

File: app/controllers/profile_controller.rb [hunk 3]
```diff
@@ -9,11 +9,9 @@ def index
     user = current_user if user.nil?
 
     # Determine how the profile should be displayed based on the user_id
-    @stories = if user == current_user
-                 paginate_stories(current_user)
-               elsif current_user.allies_by_status(:accepted).include?(user)
-                 paginate_stories(user)
-               end
+    if user == current_user || current_user.allies_by_status(:accepted).include?(user)
```
Comment (hmasila) [inline app/controllers/profile_controller.rb]: This line is too long

Comment (julianguyen) [reply app/controllers/profile_controller.rb]: You should be able to use `current_user.mutual_allies?(user)` for the allies check

**Pull request merged**